pkg/apis/serving/v1alpha1/revision_lifecycle_test.go

```diff
@@ -672,3 +673,51 @@ func TestPropagateDeploymentStatus(t *testing.T) {
 	apitest.CheckConditionSucceeded(rev.duck(), RevisionConditionResourcesAvailable, t)
 	apitest.CheckConditionSucceeded(rev.duck(), RevisionConditionContainerHealthy, t)
 }
+
+func TestPropagateAutoscalerStatus(t *testing.T) {
+	r := &RevisionStatus{}
+	r.InitializeConditions()
+	apitest.CheckConditionOngoing(r.duck(), RevisionConditionReady, t)
+
+	// PodAutoscaler has no active condition, so we are just coming up.
+	r.PropagateAutoscalerStatus(&av1alpha1.PodAutoscalerStatus{
+		Status: duckv1beta1.Status{},
+	})
+	apitest.CheckConditionOngoing(r.duck(), RevisionConditionActive, t)
+
+	// PodAutoscaler becomes ready, making us active.
+	r.PropagateAutoscalerStatus(&av1alpha1.PodAutoscalerStatus{
+		Status: duckv1beta1.Status{
+			Conditions: duckv1beta1.Conditions{{
+				Type:   av1alpha1.PodAutoscalerConditionReady,
+				Status: corev1.ConditionTrue,
+			}},
+		},
+	})
+	apitest.CheckConditionSucceeded(r.duck(), RevisionConditionActive, t)
+	apitest.CheckConditionSucceeded(r.duck(), RevisionConditionReady, t)
```

Comment:
Maybe also check for `RevisionConditionResourcesAvailable` and `RevisionConditionContainerHealthy` here?

Reply:
I'm trying to get rid of these entirely, but added a check at the end for them.